dev/Microsoft.Process.Environment/EnvironmentManager.cpp

```diff
@@ -63,4 +74,125 @@ namespace winrt::Microsoft::ProjectReunion::implementation
     {
         throw hresult_not_implemented();
     }
+
+    StringMap EnvironmentManager::GetProcessEnvironmentVariables()
+    {
+        //Get the pointer to the process block
+        LPWSTR environmentVariablesString = GetEnvironmentStrings();
+        THROW_HR_IF_NULL(E_POINTER, environmentVariablesString);
+
+        // parse the block into a map
+        LPTSTR variable;
+        variable = (LPTSTR)environmentVariablesString;
+
+        StringMap environmentVariables;
+        while (*variable)
+        {
+            std::wstring environmentVariable(variable);
```

Comment:
You're just looking to enumerate the strings in the block and split that at the '='. No need to make a string object to do the work

```
for (auto variable=environmentVariablesString; *variable; variable += wcslen(variable) + 1)
{
    auto delimiter{ wcschr(variable, L'=') };
    FAIL_FAST_HR_IF_NULL(E_UNEXPECTED, delimiter);
    std::wstring variableName(variable, 0, delimiter - variable);
    auto variableValue{ delimiter + 1 };
    environmentVariables.Insert(variableName, variableValue);
}
```

Reply:
Correct.  All I'm doing is splitting up the environment block into pairs.